From bea4ed4e9f3ecf8e4081ad820099185857fee2b5 Mon Sep 17 00:00:00 2001
From: chao zeng <chao.zeng@siemens.com>
Date: Thu, 20 Jan 2022 14:10:13 +0800
Subject: [PATCH 17/17] HACK: add delay to wait enable output stable

hardware would change the BOM to fix this issue,
currently, softwarwe to fix it

Signed-off-by: chao zeng <chao.zeng@siemens.com>
---
 board/siemens/iot2050/board.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/board/siemens/iot2050/board.c b/board/siemens/iot2050/board.c
index 74fb4aa4c1..e25d60996d 100644
--- a/board/siemens/iot2050/board.c
+++ b/board/siemens/iot2050/board.c
@@ -183,6 +183,7 @@ static void configure_serdes_by_overlays(void)
 	if(strcmp(info->name, "IOT2050-ADVANCED-M2") == 0) {
 		//Enable Power of E-key and B-key connector
 		set_pinvalue(m2_enable_power.gpio_name, m2_enable_power.label, 1);
+		udelay(4 * 1000);
         serdes_conf = get_serdes_config_set_m2_pinmux();
 		if (serdes_conf.lane0_sel == PCIE0_LANE0 && serdes_conf.lane1_sel == PCIE0_LANE1)
 			env_set("name_overlays", "ti/k3-am654-iot2050-bkey-pciex2.dtbo");
-- 
2.34.1

